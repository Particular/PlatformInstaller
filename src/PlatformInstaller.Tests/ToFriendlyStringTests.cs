using System;
using System.Text.RegularExpressions;
using ApprovalTests;
using NUnit.Framework;


[TestFixture]
public class ToFriendlyStringTests
{
    [Test]
    public void ToFriendlyString()
    {
        var currentDirectory = AssemblyLocation.CurrentDirectory.ToLowerInvariant()
            .Replace(@"bin\debug", string.Empty)
            .Replace(@"bin\release", string.Empty);
        try
        {
            ThrowException1();
        }
        catch (Exception exception)
        {
            var friendlyString = exception.ToFriendlyString().ToLowerInvariant();
            friendlyString = friendlyString
                .Replace(currentDirectory, string.Empty);
            friendlyString = new Regex(":line.*").Replace(friendlyString, "");
            Approvals.Verify(friendlyString);
        }
    }
    void ThrowException1()
    {
        ThrowException2();
    }

    void ThrowException2()
    {
        throw new Exception("Foo");
    }
}