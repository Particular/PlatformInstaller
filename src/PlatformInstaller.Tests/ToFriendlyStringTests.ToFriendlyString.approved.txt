an unhandled exception occurred:
exception:
foo
stacktrace:
   at tofriendlystringtests.throwexception2() in tofriendlystringtests.cs
   at tofriendlystringtests.throwexception1() in tofriendlystringtests.cs
   at tofriendlystringtests.tofriendlystring() in tofriendlystringtests.cs
source:
platforminstaller.tests
targetsite:
void throwexception2()
